### DISCUSSION TITLE
Export setting

### DISCUSSION THREAD

**Discussion by Vivian4849** [retorquere/zotero-better-bibtex#3115] [Q&A]
I mis-operated the export settings and delete something which is not retrivable and now my zotero cannot extract citationkey of my files in the metadata field, neither can I imprt my notes properly to obsidian. I tried to reinstall the BBT but still doesn't work. Could you please indicate how can I solve this problem? I remembered there was an option for JSON file, which was deleted by me intentionally since I didn't know what was that for. Here are the pictures of my BBT settings in Zotero. Thanks a lot!
![1](https://github.com/user-attachments/assets/190c94e3-39de-4366-8ea3-2aec96277660)
![2](https://github.com/user-attachments/assets/e46a980d-28ac-4ca6-94dc-277419caa640)

**Comment by retorquere** [comment]
> I mis-operated the export settings and delete something which is not retrivable and now my zotero cannot extract citationkey of my files in the metadata field

I don't know what this means, sorry. What metadata field?

**Reply by Vivian4849** [reply]
Thanks for reply. In my zotero, I created a formatted note where there is a table for metadata the top of each note. In the table, I set a few items such as title, author, abstract, which can be obtained by zotero automatically. However, the citation is the only item cannot be obtained. Here is a picture which I hope can help you to understand. 
![屏幕截图 2024-12-23 173709](https://github.com/user-attachments/assets/2bef7b0f-3695-4dc5-b9bf-41107d425beb)

**Reply by retorquere** [reply]
This doesn't look like Zotero? I'm not sure what role BBT plays here?

**Reply by Vivian4849** [reply]
Thanks for you reply. This is what I did with zotcard in Zotero. 
I made a note template with the code below and this is the metadata area. For other information such as title, abstract, etc, will be automatically obtained when I insert this template, however, citationkey doesn't work. Maybe it is not a problem of BBT? I am not sure. 
<h1 style="color: #2C3E50; text-align: center; border-bottom: 2px solid #616161; padding-bottom: 5px;">
  <span>📄 论文《${title}》</span>
</h1>

<h2 style="color: #FFFFFF; background-color: rgba(38, 50, 56, 0.9); padding: 10px; border-radius: 5px;">💡 Meta Data</h2>
<table style="width: 100%; border-collapse: collapse; font-size: 16px; line-height: 1.5; background-color: #212121; border: 1px solid #424242;">
  <tr> 
    <th style="background-color: #424242; text-align: right; padding: 8px; width: 20%; border: 1px solid #616161; color: #FFFFFF;">标题</th>
    <td style="background-color: #333333; padding: 8px; border: 1px solid #616161; color: #B0BEC5;">${title}</td>
  </tr>
  <tr>
    <th style="background-color: #424242; text-align: right; padding: 8px; border: 1px solid #616161; color: #FFFFFF;">期刊</th>
    <td style="background-color: #333333; padding: 8px; border: 1px solid #616161; color: #B0BEC5;">${publicationTitle}</td>
  </tr>
  <tr>
    <th style="background-color: #424242; text-align: right; padding: 8px; border: 1px solid #616161; color: #FFFFFF;">作者</th>
    <td style="background-color: #333333; padding: 8px; border: 1px solid #616161; color: #B0BEC5;">${authors}</td>
  </tr>
  <tr>
    <th style="background-color: #424242; text-align: right; padding: 8px; border: 1px solid #616161; color: #FFFFFF;">摘要</th>
    <td style="background-color: #333333; padding: 8px; border: 1px solid #616161; color: #B0BEC5;">${abstractNote}</td>
  </tr>
  <tr>
    <th style="background-color: #424242; text-align: right; padding: 8px; border: 1px solid #616161; color: #FFFFFF;">日期</th>
    <td style="background-color: #333333; padding: 8px; border: 1px solid #616161; color: #B0BEC5;">${date}</td>
  </tr>
  <tr>
    <th style="background-color: #424242; text-align: right; padding: 8px; border: 1px solid #616161; color: #FFFFFF;">DOI</th>
    <td style="background-color: #333333; padding: 8px; border: 1px solid #616161; color: #B0BEC5;">
      <a href="https://doi.org/${DOI}" style="color: #4FC3F7;">${DOI}</a>
    </td>
  </tr>
  <tr>
    <th style="background-color: #424242; text-align: right; padding: 8px; border: 1px solid #616161; color: #FFFFFF;">引用键</th>
    <td style="background-color: #333333; padding: 8px; border: 1px solid #616161; color: #B0BEC5;">${citationKey}</td>
  </tr>
</table>

<h1 style="color:  #a28ae5; padding: 10px;">🔡 KEYWORDS</h1>
<br/><hr/>

<h1 style="color: #FF4777; padding: 10px;">🎯 My Q&A</h1>
<h3 style="color: #16A085;">Q1:</h3>
<p style="color: #2C3E50; margin-left: 20px;"></p>
<h3 style="color: #E74C3C;">A1:</h3>
<p style="color: #2C3E50; margin-left: 20px;"></p>
<h3 style="color: #16A085;">Q2:</h3>
<p style="color: #2C3E50; margin-left: 20px;"></p>
<h3 style="color: #E74C3C;">A2:</h3>
<p style="color: #2C3E50; margin-left: 20px;"></p>
<br/><hr/>

<h1 style="color: #2ea8e5; padding: 10px;">💡 Relevant Insights</h1>
<br/><hr/>

<h1 style="color: #c65306; padding: 10px;">🔬 Gap</h1>
<br/><hr/>

<h1 style="color: #ff6666; padding: 10px;">⏭️ RFW</h1>
<br/><hr/>

<h1 style="color: #fc8c23; padding: 10px;">🌟 Contributions</h1>
<br/><hr/>

<h1 style="color: #1b54f2; padding: 10px;">🪢 Limitation</h1>
<br/><hr/>

<h1 style="color: #9B59B6; padding: 10px;">😇 Reflection</h1>
<br/><hr/>

<h1 style="color: #2C3E50; padding: 10px;">🔗 Citation</h1>
<br/><hr/>

**Reply by retorquere** [reply]
You'd have to ask the zotcard developer. I can explain to the dev how to get the key if that is unclear but I don't know how zotcard populates notes.
